src/main/java/com/zombiecastlerush/entity/Role.java

```diff
@@ -93,52 +90,44 @@ public void setHealth(int points) {
         }
     }
 
-    public String getName() {
-        return name;
-    }
-
-    public void setName(String name) {
-        this.name = name;
-    }
-
     public Inventory getInventory() {
         return inventory;
     }
 
     public void setInventory(Inventory inventory) {
         this.inventory = inventory;
     }
-        // Inventory methods
-        public Item pickUp (Item item){
-            for (Item existingItem : this.getCurrentPosition().inventory.getItems()) {
-                System.out.println(existingItem.getName());
-                if (item.equals(existingItem)) {
-                    this.inventory.addItems(item);
-                    this.getCurrentPosition().inventory.deleteItems(item);
-                    System.out.println(item.getName() + " picked up by " + this.getName());
-                    return item;
-                }
+    // Inventory methods
+    public Item pickUp (Item item){
+        for (Item existingItem : this.getCurrentPosition().inventory.getItems()) {
+            System.out.println(existingItem.getName());
+            if (item.equals(existingItem)) {
+                this.inventory.addItems(item);
+                this.getCurrentPosition().inventory.deleteItems(item);
+                System.out.println(item.getName() + " picked up by " + this.getName());
+                return item;
             }
-            return null;
         }
+        return null;
+    }
 
-        public Item drop (Item item){
-            for (Item existingItem : this.inventory.getItems()) {
-                if (item.equals(existingItem)) {
-                    this.getCurrentPosition().inventory.addItems(item);
-                    this.inventory.deleteItems(item);
-                    System.out.println(item.getName() + " dropped by " + this.getName());
-                    return item;
-                }
+    public Item drop (Item item){
+        for (Item existingItem : this.inventory.getItems()) {
+            if (item.equals(existingItem)) {
+                this.getCurrentPosition().inventory.addItems(item);
+                this.inventory.deleteItems(item);
+                System.out.println(item.getName() + " dropped by " + this.getName());
+                return item;
             }
-            return null;
         }
+        return null;
+    }
 
-        public void dropAll(){
-            for (Item item : this.inventory.getItems()) {
-                this.getCurrentPosition().inventory.addItems(item);
-            }
-            this.inventory.deleteAllItems();
+    public void dropAll(){
```

Comment:
Does Puzzle and Challenge need this dropAll() too? if that is what we want, we can have this in Entity like dropAll(Entity destination)

Reply:
It's a good point, but no great solution right now. Will think about it.